src/client/clientenvironment.cpp

```diff
@@ -362,26 +362,26 @@ GenericCAO* ClientEnvironment::getGenericCAO(u16 id)
 	return NULL;
 }
 
-u16 ClientEnvironment::addActiveObject(ClientActiveObject *object)
+u16 ClientEnvironment::addActiveObject(std::unique_ptr<ClientActiveObject> object)
 {
+	auto obj = object.get();
```

Comment:
aren't we defeating the std::unique_ptr with this method, where we move the unique_ptr then we always have the pointer here and we are playing with it ?

Reply:
I'm not sure if I understand the question correctly.

`obj` is a non-owning ref/ptr. `object` owns. After the `registerObject` call, someone else owns the object, but references (i.e. `obj`) are still valid (at least if `registerObject` returns true).
(Before, there was just the raw ptr `object`. It was an owning raw ptr when we got it, but after the `registerObject` call it suddenly was no longer owning.)
The point of unique ptrs is not that you don't use other references on the object anymore.